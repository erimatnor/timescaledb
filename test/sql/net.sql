\c single :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_parsing(int) RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_parsing' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_parsing_full() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_parsing_full' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_request_build() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_request_build' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_conn() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_conn' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_parse_uri(text) RETURNS
    TABLE(scheme text, host text, port int, path text)
    AS :MODULE_PATHNAME, 'test_parse_uri' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;

\c single :ROLE_DEFAULT_PERM_USER
SELECT _timescaledb_internal.test_http_parsing(10000);
SELECT _timescaledb_internal.test_http_parsing_full();
SELECT _timescaledb_internal.test_http_request_build();
SELECT _timescaledb_internal.test_conn();

SELECT _timescaledb_internal.test_parse_uri('http://foo.bar.com');
SELECT _timescaledb_internal.test_parse_uri('http://foo.bar.com:8080');
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com');
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com:8443');
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com/v1/telemetry');
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com/v1/telemetry/');
\set ON_ERROR_STOP false
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com:8443?');
SELECT _timescaledb_internal.test_parse_uri('foo://foo.bar.com');
SELECT _timescaledb_internal.test_parse_uri('http//foo.bar.com');
SELECT _timescaledb_internal.test_parse_uri('http:/foo.bar.com');
