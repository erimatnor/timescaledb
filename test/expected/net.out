\c single :ROLE_SUPERUSER
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_parsing(int) RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_parsing' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_parsing_full() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_parsing_full' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_http_request_build() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_http_request_build' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_conn() RETURNS VOID
    AS :MODULE_PATHNAME, 'test_conn' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
CREATE OR REPLACE FUNCTION _timescaledb_internal.test_parse_uri(text) RETURNS
    TABLE(scheme text, host text, port int, path text)
    AS :MODULE_PATHNAME, 'test_parse_uri' LANGUAGE C IMMUTABLE STRICT PARALLEL SAFE;
\c single :ROLE_DEFAULT_PERM_USER
SELECT _timescaledb_internal.test_http_parsing(10000);
 test_http_parsing 
-------------------
 
(1 row)

SELECT _timescaledb_internal.test_http_parsing_full();
 test_http_parsing_full 
------------------------
 
(1 row)

SELECT _timescaledb_internal.test_http_request_build();
 test_http_request_build 
-------------------------
 
(1 row)

SELECT _timescaledb_internal.test_conn();
 test_conn 
-----------
 
(1 row)

SELECT _timescaledb_internal.test_parse_uri('http://foo.bar.com');
     test_parse_uri     
------------------------
 (http,foo.bar.com,80,)
(1 row)

SELECT _timescaledb_internal.test_parse_uri('http://foo.bar.com:8080');
      test_parse_uri      
--------------------------
 (http,foo.bar.com,8080,)
(1 row)

SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com');
      test_parse_uri      
--------------------------
 (https,foo.bar.com,443,)
(1 row)

SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com:8443');
      test_parse_uri       
---------------------------
 (https,foo.bar.com,8443,)
(1 row)

SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com/v1/telemetry');
            test_parse_uri             
---------------------------------------
 (https,foo.bar.com,443,/v1/telemetry)
(1 row)

SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com/v1/telemetry/');
             test_parse_uri             
----------------------------------------
 (https,foo.bar.com,443,/v1/telemetry/)
(1 row)

\set ON_ERROR_STOP false
SELECT _timescaledb_internal.test_parse_uri('https://foo.bar.com:8443?');
ERROR:  URI does not support query parameters
SELECT _timescaledb_internal.test_parse_uri('foo://foo.bar.com');
ERROR:  invalid URI scheme
SELECT _timescaledb_internal.test_parse_uri('http//foo.bar.com');
ERROR:  invalid URI
SELECT _timescaledb_internal.test_parse_uri('http:/foo.bar.com');
ERROR:  invalid URI
